### PR TITLE
BC - Updates for Laravel 10 and the MonologHandler

### PR DESCRIPTION
**Pull request by michaelscola**
Updated the composer.json file to be inline with Laravel 10 - dropped all PHP support except 8.1 and added illuminate/support 10 only

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,8 +22,8 @@
         }
     ],
     "require": {
-        "php": ">=7.2|^8.0",
-        "illuminate/support": "^6.0|^7.0|^8.0|^9.0|^10.0",
+        "php": "^8.1",
+        "illuminate/support": "^10.0",
         "rollbar/rollbar": "^2.0 | ^3.1"
```
Comment (danielmorell) [inline composer.json:27]: I think we may want to drop support for `rollbar/rollbar` v2 as it is not compatible with PHP 8.

**Comment by danielmorell** [comment]
This is related to #141.

File: composer.json (hunk 1 above)
Comment (jonnott) [reply composer.json:27]: ..and you're right it's not possible to support Laravel 9.x and below, as they only support monolog v2 and not v3.

**Comment by jonnott** [comment]
NB @michaelscola Have you seen Monolog's upgrade guide, where is hints at how you'd possibly support multiple Monolog versions (e.g. 2/3) in terms of LogRecord vs array arguments? https://github.com/Seldaek/monolog/blob/main/UPGRADE.md

**Comment by michaelscola** [comment]
> NB @michaelscola Have you seen Monolog's upgrade guide, where it hints at how you'd possibly support multiple Monolog versions (e.g. 2/3) in terms of LogRecord vs array arguments? https://github.com/Seldaek/monolog/blob/main/UPGRADE.md

I did, but you are going to run into an issue because older PHP versions don't support union types and thus will break our typehinting. So really the only option I see is a BC.

**Comment by jonnott** [comment]
> I did, but you are going to run into an issue because older PHP versions don't support union types and thus will break our typehinting. So really the only option I see is a BC.

Sure .. easiest to just target Monolog 3, PHP 8.1, and Laravel 10 then.

File: composer.json (hunk 1 above)
Comment (michaelscola) [reply composer.json]: > I think we may want to drop support for `rollbar/rollbar` v2 as it is not compatible with PHP 8.

Removed older 2x

**Comment by michaelscola** [comment]
> > I did, but you are going to run into an issue because older PHP versions don't support union types and thus will break our typehinting. So really the only option I see is a BC.
> 
> Sure .. easiest to just target Monolog 3, PHP 8.1, and Laravel 10 then.

It does seem like a path of diminishing returns to try and get it working.

**Comment by jonnott** [comment]
@michaelscola See https://github.com/rollbar/rollbar-php/pull/602

**Comment by danielmorell** [comment]
We released `v4.0.0-rc` of the core `rollbar/rollbar` library today! It supports Monolog 3.

@michaelscola Could you bump `rollbar/rollbar` to `^4.0`. I also think `orchestra/testbench` should be upgraded to v8.

After that I will make the needed updates to automated test suite, then I think we should be in a position to release a beta version for testing.

**Comment by michaelscola** [comment]
I will go ahead and get this updated today.

**Comment by michaelscola** [comment]
> We released `v4.0.0-rc` of the core `rollbar/rollbar` library today! It supports Monolog 3.
> 
> @michaelscola Could you bump `rollbar/rollbar` to `^4.0`. I also think `orchestra/testbench` should be upgraded to v8.
> 
> After that I will make the needed updates to automated test suite, then I think we should be in a position to release a beta version for testing.

This has been completed

**Review by danielmorell — APPROVED** [review]
This looks great! Thank you @michaelscola.

**Comment by danielmorell** [comment]
I will work on those CI tests this afternoon.

**Pull request merged**